ref-exchange/src/simple_pool.rs

```diff
@@ -182,7 +183,8 @@ impl SimplePool {
             result.push(amount);
         }
         if prev_shares_amount == shares {
-            self.shares.remove(&sender_id);
+            // [AUDIT_13] never unregister a LP when he remove liqudity.
+            self.shares.insert(&sender_id, &0);
```

Comment:
let's add some unit tests that would
 a) fail with previous solution
 b) fail if this string is commented

Reply:
OK